src/SunburstVisual.ts

```diff
@@ -0,0 +1,443 @@
+                this.selectedCategoryLabel.classed(this.appCssConstants.labelVisible.class, this.settings.group.showSelected);
+                this.calculateLabelPosition();
+            }
+        }
+        private visualHost: IVisualHost;
+        private rawData: DataViewHierarchy;
+        private data: SunburstData;
+        private arc: d3.svg.Arc<SunburstSlice>;
+        private svg: d3.Selection<{}>;
+        private main: d3.Selection<{}>;
+        private percentageLabel: d3.Selection<string>;
+        private selectedCategoryLabel: d3.Selection<string>;
+
+        private appCssConstants: IAppCssConstants = {
+            main: createClassAndSelector("sunburst"),
+            mainInteractive: createClassAndSelector("sunburst--interactive"),
+            slice: createClassAndSelector("sunburst__slice"),
+            sliceSelected: createClassAndSelector("sunburst__slice--selected"),
+            label: createClassAndSelector("sunburst__label"),
+            labelVisible: createClassAndSelector("sunburst__label--visible"),
+            categoryLabel: createClassAndSelector("sunburst__category-label"),
+            percentageLabel: createClassAndSelector("sunburst__percentage-label")
+        };
+        private colors: IColorPalette;
+        private selectionManager: ISelectionManager;
+        private tooltipService: ITooltipServiceWrapper;
+        constructor(options: VisualConstructorOptions) {
+            this.visualHost = options.host;
+            this.tooltipService = createTooltipServiceWrapper(
+                options.host.tooltipService,
+                options.element);
+
+            this.arc = d3.svg.arc<SunburstSlice>()
+                .startAngle((slice: SunburstSlice) => slice.x)
+                .endAngle((slice: SunburstSlice) => slice.x + slice.dx)
+                .innerRadius((slice: SunburstSlice) => Math.sqrt(slice.y))
+                .outerRadius((slice: SunburstSlice) => Math.sqrt(slice.y + slice.dy));
+
+            this.colors = options.host.colorPalette;
+            this.selectionManager = options.host.createSelectionManager();
+
+            this.svg = d3.select(options.element)
+                .append("svg")
+                .attr("viewBox", `0 0 ${Sunburst.ViewBoxSize}  ${Sunburst.ViewBoxSize}`)
+                .attr("width", "100%")
+                .attr("height", "100%")
+                .attr("preserveAspectRatio", "xMidYMid meet")
+                .classed(this.appCssConstants.main.class, true);
+
+            this.main = this.svg.append("g");
+            this.main.attr(CssConstants.transformProperty, translate(Sunburst.CentralPoint, Sunburst.CentralPoint));
+
+            this.selectedCategoryLabel = <d3.Selection<string>>this.svg
+                .append("text")
+                .classed(this.appCssConstants.label.class, true)
+                .classed(this.appCssConstants.categoryLabel.class, true);
+            this.selectedCategoryLabel.attr("text-anchor", "middle");
```

Comment:
I'm not so sure regarding you statement. Please take a look at [this](https://developer.mozilla.org/en/docs/Web/SVG/Attribute/text-anchor) document.

```
As a presentation attribute, it also can be used as a property directly inside a CSS stylesheet.
```